### PR TITLE
feat: Adding bigquery vector search notebook

### PR DESCRIPTION
**Pull request by sethijaideep**
# Description
Added a notebook to showcase BigQuery vector search capability. 

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Follow the [`CONTRIBUTING` Guide](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/CONTRIBUTING.md).
- [x] You are listed as the author in your notebook or README file.
  - [x] Your account is listed in [`CODEOWNERS`](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS) for the file(s).
- [x] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [x] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format).
- [x] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+        "    <a href=\"https://console.cloud.google.com/vertex-ai/colab/import/https:%2F%2Fraw.githubusercontent.com%2FGoogleCloudPlatform%2Fgenerative-ai%2Fmain%2Fgemini%2Fuse-cases%2Fapplying-llms-to-data%2Fsemantic-search-in-bigquery%2Fstackoverflow_questions_semantic_search.ipynb\">\n",
+        "      <img width=\"32px\" src=\"https://lh3.googleusercontent.com/JmcxdQi-qOpctIvWKgPtrzZdJJK-J3sWE1RsfjZNwshCFgE_9fULcNpuXYTilIR2hjwN\" alt=\"Google Cloud Colab Enterprise logo\"><br>Run in Colab Enterprise\n",
+        "    </a>\n",
+        "  </td>\n",
+        "  <td style=\"text-align: center\">\n",
+        "    <a href=\"https://github.com/GoogleCloudPlatform/generative-ai/blob/main/gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb\">\n",
+        "      <img src=\"https://cloud.google.com/ml-engine/images/github-logo-32px.png\" alt=\"GitHub logo\"><br>View on GitHub\n",
+        "    </a>\n",
+        "  </td>\n",
+        "</table>"
```
Comment (holtskinner) [inline gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb:45]: Please use the full table from [notebook_template.ipynb](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/notebook_template.ipynb)

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "HKNY2cVBUB9r"
+      },
+      "source": [
+        "## How to open this notebook in BigQuery Studio\n",
+        "\n",
+        "This notebook was written to be compatible for use within BigQuery Studio. To open this notebook, click to [Open in Colab Enterprise](https://console.cloud.google.com/vertex-ai/colab/import/https:%2F%2Fraw.githubusercontent.com%2FGoogleCloudPlatform%2Fgenerative-ai%2Fmain%2Fgemini%2Fuse-cases%2Fapplying-llms-to-data%2Fsemantic-search-in-bigquery%2Fstackoverflow_questions_semantic_search.ipynb). This will open a new window in the Cloud Console and prompt you to confirm import. Then, navigate to **BigQuery**, where you will find the notebook available in the **Explorer pane** under **Notebooks**."
```
Comment (holtskinner) [inline gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb:85]: Can you remove this section with the Colab link or just direct them to use the table at the top?

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+        "## Verify vector index creation"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "v-MJ0Gtui_BZ"
+      },
+      "source": [
+        "Note : The vector index is populated asynchronously. You can check whether the index is ready to be used by querying the INFORMATION_SCHEMA.VECTOR_INDEXES view and verifying that the coverage_percentage column value is greater than 0 and the last_refresh_time column value isn't NULL."
```
Comment (holtskinner) [inline gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb:355]:
```suggestion
        "Note : The vector index is populated asynchronously. You can check whether the index is ready to be used by querying the `INFORMATION_SCHEMA.VECTOR_INDEXES` view and verifying that the `coverage_percentage` column value is greater than 0 and the `last_refresh_time` column value isn't `NULL`."
```

Do similar for any "code-like" value in markdown

Comment (sethijaideep) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb]: Done

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb (hunk 1 above)
Comment (sethijaideep) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb]: Done, added the table from the sample notebook.

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 4]
```diff
@@ -0,0 +1,484 @@
+{
```
Comment (jeffonelson) [inline gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb:1]: <p>Add code markdown formatting around VECTOR_SEARCH</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/1135/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+        "      ML.GENERATE_EMBEDDING( MODEL `bigquery_demo.text_embedding_004`,\n",
+        "        (\n",
+        "        SELECT\n",
+        "          'Iphone app using RSS is crashing' AS content) ) ),\n",
+        "    top_k => 5,\n",
+        "    OPTIONS => '{\"fraction_lists_to_search\": 0.10}')\n",
+        "ORDER BY\n",
+        "  distance ASC ;"
+      ]
+    },
```
[the remark this answers is not in the document]
Comment (sethijaideep) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb:430]: good suggestions, i have limited the input question set to top 10,000 posts specific to ios .

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+        "    (\n",
+        "    SELECT\n",
+        "      id,\n",
+        "      title,\n",
+        "      body,\n",
+        "      CONCAT (title, body ) AS CONTENT\n",
+        "    FROM\n",
+        "      `bigquery-public-data.stackoverflow.posts_questions`\n",
+        "    LIMIT\n",
+        "      10000 ),\n",
```
[the remark this answers is not in the document]
Comment (sethijaideep) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb:298]: Good check, I have dropped the record limit to 10K .  BQ Scan bytes is around 36 GB , which is still under the Free tier limit of On-Demand billing model. Batch text embedding cost is below 1$ mark.

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 7]
```diff
@@ -0,0 +1,484 @@
...
+      "metadata": {
+        "id": "_cGtn8TvG7SB"
+      },
+      "source": [
+        "## Overview\n",
+        "\n",
+        "The objective is to demonstrate how to perform semantic search in BigQuery using Vector Search, including:\n",
+        "\n",
+        "\n",
+        "*   Completing setup steps for accessing Vertex AI from BigQuery\n",
```
[the remark this answers is not in the document]
Comment (sethijaideep) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb:80]: I have made the verb tense consistent.

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 8]
```diff
@@ -0,0 +1,484 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "sf3ioTDBzuHR"
+      },
+      "source": [
+        "## About the dataset\n",
+        "\n",
+        "We are going to use Stack Overflow public dataset available in BigQuery . The data is an archive of Stack Overflow posts, votes, tags and badges.\n",
```
[the remark this answers is not in the document]
Comment (sethijaideep) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb]: Thanks, done.

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 9]
```diff
@@ -0,0 +1,484 @@
...
+      "execution_count": null,
+      "metadata": {
+        "id": "65046bc876c6"
+      },
+      "outputs": [],
+      "source": [
+        "!bq mk --connection --location=us \\\n",
+        "    --connection_type=CLOUD_RESOURCE vertex_conn"
+      ]
+    },
```
[the remark this answers is not in the document]
Comment (sethijaideep) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb]: Added link to cloud resource connection doc space.

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 10]
```diff
@@ -0,0 +1,484 @@
...
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "xCbjG3fb3def"
+      },
+      "source": [
+        "# Create the remote model in BigQuery ML"
+      ]
+    },
```
[the remark this answers is not in the document]
Comment (sethijaideep) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb:210]: added some text, please feel free to suggest changes.

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 11]
```diff
@@ -0,0 +1,484 @@
...
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "JcYVG4qNKAuY"
+      },
+      "source": [
+        "# Prepare the dataset for semantic search"
+      ]
+    },
```
[the remark this answers is not in the document]
Comment (sethijaideep) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb]: added

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 12]
```diff
@@ -0,0 +1,484 @@
...
+      "execution_count": null,
+      "metadata": {
+        "id": "8bcyGsNWHPeZ"
+      },
+      "outputs": [],
+      "source": [
+        "%%bigquery\n",
+        "SELECT * FROM `bigquery_demo.posts_questions_embedding` LIMIT 100;"
+      ]
+    },
```
[the remark this answers is not in the document]
Comment (sethijaideep) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb:337]: added

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb [hunk 13]
```diff
@@ -0,0 +1,484 @@
...
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ceUN_Mgm0vQa"
+      },
+      "source": [
+        "## Create Vector Index on the embeddings to help with efficient semantic search"
+      ]
+    },
```
[the remark this answers is not in the document]
Comment (sethijaideep) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb]: Added link and context on vector index

File: gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb (hunk 11 above)
[the remark this answers is not in the document]
Comment (jeffonelson) [reply gemini/use-cases/applying-llms-to-data/semantic-search-in-bigquery/stackoverflow_questions_semantic_search.ipynb]: Will follow up on this.

**Pull request merged**